test/tools/router_check/router.cc

```diff
@@ -265,8 +267,12 @@ bool RouterCheckTool::compareRewritePath(ToolConfig& tool_config, const std::str
   Envoy::StreamInfo::StreamInfoImpl stream_info(Envoy::Http::Protocol::Http11,
                                                 factory_context_->dispatcher().timeSource());
   if (tool_config.route_->routeEntry() != nullptr) {
-    tool_config.route_->routeEntry()->finalizeRequestHeaders(*tool_config.headers_, stream_info,
-                                                             true);
+    if (!perform_finalize_headers_) {
+      tool_config.route_->routeEntry()->finalizeRequestHeaders(*tool_config.headers_, stream_info,
+                                                               true);
+      perform_finalize_headers_ = true;
```

Comment:
nit: it looks like the bool you added is indicating that the finalization has occurred if it is true. Maybe rename to `headers_finalized_` , indicating a state, whereas `perform_finalize_headers_` is a verb?

Reply:
Thanks. Took care of it now.